airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -44,9 +38,41 @@ public StandardSyncOutput run(final JobRunConfig jobRunConfig,
                                 final StandardSyncInput syncInput,
                                 final UUID connectionId) {
 
-    StandardSyncOutput syncOutput = replicationActivity.replicate(jobRunConfig, sourceLauncherConfig, destinationLauncherConfig, syncInput);
     final int version = Workflow.getVersion(VERSION_LABEL, Workflow.DEFAULT_VERSION, CURRENT_VERSION);
 
+    final ReplicationActivity replicationActivity;
+    final NormalizationActivity normalizationActivity;
+    final DbtTransformationActivity dbtTransformationActivity;
+    final PersistStateActivity persistActivity;
+
+    /**
+     * The current version calls a new activity to determine which Task Queue to use for other
+     * activities. The previous version doesn't call this new activity, and instead lets each activity
+     * inherit the workflow's Task Queue.
+     */
+    if (version > PREV_VERSION) {
```

Comment:
Given the changes we made to error handling in temporal, do we need this? I thought one benefit of catching the non-determinism error would be that we wouldn't have to play this versioning game. @evantahler

Reply:
Continued in child PR here: https://github.com/airbytehq/airbyte/pull/15905/files#r953199607